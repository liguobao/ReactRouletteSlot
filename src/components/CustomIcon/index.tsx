/*
 * @Author: wzi
 * @Date: 2018-04-10 11:30:47
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-18 11:51:47
 */
import React from 'react';
import styled from '@common/helper/styled-component';
type CustomIconProp = {
    type: string;
    className?: string;
    size?: string;
    onClick?: any;
};

const CustomIcons: React.StatelessComponent<CustomIconProp> = ({
    type,
    className = '',
    ...restProps
}) => (
    <svg className={`icon ${className}`} aria-hidden={true} {...restProps}>
        <use xlinkHref={`#icon-${type}`} />
    </svg>
);
const CustomIcon = styled(CustomIcons)`
    width: ${(props) => (props.size ? props.size : '1.5em')} !important;
    height: ${(props) => (props.size ? props.size : '1.5em')} !important;
`;

export default CustomIcon;
