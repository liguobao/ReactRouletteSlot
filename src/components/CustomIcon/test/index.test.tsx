/*
 * @Author: wzi
 * @Date: 2018-01-31 16:40:02
 * @Last Modified by: wzi
 * @Last Modified time: 2018-04-25 17:19:28
 */
import React from 'react';
import CustomIcon from '@components/CustomIcon';
import { mount } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
describe('CustomIcon', () => {
    let _props, _spies, _wrapper;
    beforeEach(() => {
        _spies = {};
        _props = {
            onClick: (_spies.onClick = jest.fn(() => {
                console.log('Icon被点击了');
            })),
            type: 'test',
            size: null,
            className: 'haha',
        };
        _wrapper = mount(<CustomIcon {..._props} />);
    });
    test('整体', () => {
        expect(toJson(_wrapper)).toMatchSnapshot();
        expect(_wrapper.find('use').prop('xlinkHref')).toBe(
            `#icon-${_props.type}`
        );
    });
    test('无 size', () => {
        expect(_wrapper).toHaveStyleRule('width', '1.5em !important');
        expect(_wrapper).toHaveStyleRule('height', '1.5em !important');
    });
    test('设置 size', () => {
        _wrapper.setProps({ size: '100px' });
        expect(_wrapper).toHaveStyleRule('width', '100px !important');
        expect(_wrapper).toHaveStyleRule('height', '100px !important');
    });
    test('设置 className', () => {
        _wrapper.setProps({ className: 'test' });
        expect(_wrapper.prop('className')).toContain('test');
    });
});
