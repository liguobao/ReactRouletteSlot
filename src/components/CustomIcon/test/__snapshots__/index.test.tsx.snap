// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomIcon 整体 1`] = `
.c0 {
  width: 1.5em !important;
  height: 1.5em !important;
  margin-top: 5px;
}

<Styled(CustomIcon)
  className="haha"
  onClick={[MockFunction]}
  size={null}
  type="test"
>
  <CustomIcon
    className="haha c0"
    onClick={[MockFunction]}
    size={null}
    type="test"
  >
    <svg
      aria-hidden={true}
      className="icon haha c0"
      onClick={[MockFunction]}
      size={null}
    >
      <use
        xlinkHref="#icon-test"
      />
    </svg>
  </CustomIcon>
</Styled(CustomIcon)>
`;
